yourgov/components/FormMobileFoodVendorPermit/FormTask1FormState.ts

```diff
@@ -0,0 +1,156 @@
+import { DeepPartial } from 'react-hook-form';
+import * as yup from 'yup';
+import { yupDateField, yupPhoneField } from './utils';
+
+export const task1Step1FormSchema = yup
+	.object({
+		firstName: yup.string().required('Enter your first name'),
+		lastName: yup.string().required('Enter your last name'),
+		email: yup
+			.string()
+			.email('Enter a valid email')
+			.required('Enter your email'),
+		contactPhoneNumber: yupPhoneField.optional(),
+	})
+	.required();
+
+export type Task1Step1FormSchema = yup.InferType<typeof task1Step1FormSchema>;
+
+export const task1Step1Part2FormSchema = yup
+	.object({
+		contactPhoneNumber: yupPhoneField.optional(),
+	})
+	.required();
+
+export type Task1Step1Part2FormSchema = yup.InferType<
+	typeof task1Step1Part2FormSchema
+>;
+
+export const task1Step2FormSchema = yup
+	.object({
+		businessName: yup.string().required('Business or company name is required'),
+		tradingName: yup.string(),
+		businessStructure: yup
+			.string()
+			.typeError('Business structure is required')
+			.required('Business structure is required'),
+		abn: yup.string().when('businessStructure', (value, schema) => {
+			if (value === 'Business') {
+				return schema.required('ABN is required');
```

Comment:
This error shouldn't show if you haven't selected anything but have other errors.

Steps to repro:

1. Go to this step
2. Click Save and continue
3. Receive errors
4. Click Business radio button
5. ABN error shouldn't be shown

Reply:
I'm not sure if there's anything we can do about this. The default behaviour of react-hook-form is to validate on submit and then to validate on change. In your reproduction steps, step 4 would trigger an onChange of the form so the validation would trigger again. Let's have a call about this.

Reply:
To discuss later outside of this PR.